Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer.swift

```diff
@@ -244,7 +250,8 @@ extension NIOAsyncSequenceProducer {
 
         /// The result of a call to ``NIOAsyncSequenceProducer/Source/yield(_:)``.
         public enum YieldResult: Hashable {
-            /// Indicates that the caller should produce more elements.
+            /// Indicates that the caller should produce more elements for now. The delegate's ``NIOAsyncSequenceProducerDelegate/produceMore()``
+            /// method will get called once production should be resumed.
             case produceMore
             /// Indicates that the caller should stop producing elements.
```

Comment:
```suggestion
            /// Indicates that the caller should produce more elements for now. The delegate's ``NIOAsyncSequenceProducerDelegate/produceMore()`` will not get called, since the demand was already signaled through this ``YieldResult``
            case produceMore
            /// Indicates that the caller should stop producing elements. The delegate's ``NIOAsyncSequenceProducerDelegate/produceMore()``
            /// method will get called once production should be resumed.
```